src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -288,20 +302,256 @@ private static ProjectItemGroupElement CreateItemGroup(Project project, string f
             }
         }
 
+        internal Dictionary<string, Tuple<IEnumerable<PRPackage>, IEnumerable<PRPackage>>> GetPackageReferencesForList(
+            Project project, IList<string> userInputFrameworks, bool transitive)
+        {
+            //Path to the assets file
+            var assetsPath = project.GetPropertyValue(ASSETS_FILE_PATH_TAG);
+
+            //If the property or file were not found, print an error message and return
+            if (assetsPath == "" || !File.Exists(assetsPath))
+            {
+                Console.Error.WriteLine(string.Format(CultureInfo.CurrentCulture,
+                    Strings.Error_AssetsFileNotFound,
+                    project.FullPath));
+                return null;
+            }
+
+
+            var lockFileFormat = new LockFileFormat();
+            var assetsFile = lockFileFormat.Read(assetsPath);
+
+            //Packages from the assets file
+            var requestedVersions = GetRequestedPackages(project, userInputFrameworks, assetsFile);
+
+            //Packages resolved versions from the assets file including transitive
+            //packages if include-transitive flag is used
+            var packagesPerFramework = GetResolvedVersions(userInputFrameworks, assetsFile, requestedVersions, transitive);
+
+            return packagesPerFramework;
+        }
+
+        /// <summary>
+        /// Given a project, a list of frameworks and an assets file, this function returns all the
+        /// target frameworks with their packages
+        /// </summary>
+        /// <param name="project">PR project file</param>
+        /// <param name="userInputFrameworks">List of frameworks</param>
+        /// <param name="assetsFile">Assets file</param>
+        /// <returns></returns>
+        private Dictionary<string, IEnumerable<PRPackage>> GetRequestedPackages(Project project, IList<string> userInputFrameworks, LockFile assetsFile)
+        {
+            var result = new Dictionary<string, IEnumerable<PRPackage>>();
+
+            foreach (var tfm in assetsFile.PackageSpec.TargetFrameworks)
+            {
+                if (userInputFrameworks.Count == 0 || userInputFrameworks.Contains(tfm.ToString())) //If tfm is in the list given by the user or no frameworks were specified
+                {
+                    var projPackages = GetPackageReferencesPerFramework(project, "", tfm.ToString());
```

Comment:
why do you need a separate api to get packageReference. Assets file will give you everything you need here pretty straightforward... I dont understand why you really need to access MSBuild project here?